pkg/operator/terminationobserver/termination_metrics.go

```diff
@@ -0,0 +1,57 @@
+package terminationobserver
+
+import (
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+type terminationMetrics struct {
+	store           Store
+	apiServerMetric *prometheus.GaugeVec
+}
+
+// RegisterMetrics registers the termination metrics into legacy prometheus registry.
+func RegisterMetrics(store Store) {
+	legacyregistry.MustRegister(newTerminationMetrics(store))
+}
+
+func newTerminationMetrics(store Store) *terminationMetrics {
+	return &terminationMetrics{
+		store: store,
+		apiServerMetric: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "openshift_kube_apiserver_termination",
+			Help: "Report individual termination events and their timestamps for each Kubernetes API server",
+		}, []string{"name", "createdTimestamp", "eventName"}),
```

Comment:
isn't a timestamp as label exploding memory consumption for a gauge?

Reply:
verified with metrics team: timestamp labels are a big NO.

Reply:
https://i.imgflip.com/3i9htp.jpg with regards from monitoring team.

Reply:
We can use the unix timestamp as value and if it is == 0 then it means we haven't seen that event. I'm not sure how will the prom query looks like then to plot a timeline that shows the timeline of termination for particular API server